src/librustc_target/abi/mod.rs

```diff
@@ -828,12 +828,13 @@ pub enum Variants {
         index: VariantIdx,
     },
 
-    /// Enums with more than one inhabited variant: for each case there is
-    /// a struct, and they all have space reserved for the discriminant,
-    /// which is the sole field of the enum layout.
+    /// Enum-likes with more than one inhabited variant: for each case there is
+    /// a struct, and they all have space reserved for the discriminant.
+    /// For enums this is the sole field of the layout.
     Multiple {
         discr: Scalar,
         discr_kind: DiscriminantKind,
+        discr_index: usize,
```

Comment:
also use https://github.com/rust-lang/rust/blob/1d4f0e71c6cd55d43f1d7b9b58cab8ef7c6f4117/src/librustc/mir/mod.rs#L2004 here

Reply:
We're in `rustc_target`. Also, this whole module uses `usize` (if not `u32`) for field indices.

Reply:
yea, this doesn't have to happen in this PR, and we can move the `Field` definition up into rustc_target